cclib/method/volume.py

```diff
@@ -178,23 +178,28 @@ def writeasvtk(self, filename):
         )
         v.tofile(filename)
 
-    def integrate(self):
+    def integrate(self, weights=None):
+        weights = numpy.ones_like(self.data) if weights is None else weights

```

Comment:
can we assert that weights is the right size

Reply:
Yes -- I'll add it in. Thanks!